src/main/java/teammates/storage/search/SearchQuery.java

```diff
@@ -40,14 +43,31 @@ public int getFilterSize(){
     }
     
     protected SearchQuery setTextFilter(String textField, String queryString){
-        String sanitizedQueryString = Sanitizer.sanitizeForSearch(queryString).toLowerCase().trim();
+        String sanitizedQueryString;
+        if(isValidEmailAddress(queryString)){
+            sanitizedQueryString = queryString.toLowerCase().trim();
+        } else {
+            sanitizedQueryString = Sanitizer.sanitizeForSearch(queryString).toLowerCase().trim(); 
+        }
+        
         if(!sanitizedQueryString.isEmpty()){
             String preparedOrQueryString = prepareOrQueryString(sanitizedQueryString);
             this.textQueryStrings.add(textField + ":" + preparedOrQueryString);
         }
         return this;
     }
     
+    private boolean isValidEmailAddress(String email) {
+        boolean result = true;
+        try {
+           InternetAddress emailAddress = new InternetAddress(email);
+           emailAddress.validate();
+        } catch (AddressException exception) {
+           result = false;
+        }
```

Comment:
Note that it is generally not advisable to use exceptions for normal work such as figuring out if an email address is correct. Exceptions are for exceptional situations. But it's ok if there is no other way.